testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java

```diff
@@ -675,8 +696,26 @@ public void testDelete()
 
         assertUpdate("DROP TABLE " + tableName);
 
-        // delete with multiple SemiJoin
+        // delete using a scalar and EXISTS subquery
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM orders", "SELECT count(*) FROM orders");
+        assertUpdate("DELETE FROM " + tableName + " WHERE orderkey = (SELECT orderkey FROM orders ORDER BY orderkey LIMIT 1)", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE orderkey = (SELECT orderkey FROM orders WHERE false)", 0);
```

Comment:
Can we add support for correlated sub query ... So that we could ensure it works for corelated joins too.

Reply:
It is out of the scope of this PR. Surely it won't be supported for jdbc connectors.